hypertrace-trace-enricher/src/main/java/org/hypertrace/traceenricher/trace/enricher/StructuredTraceEnrichProcessor.java

```diff
@@ -1,34 +1,40 @@
 package org.hypertrace.traceenricher.trace.enricher;
 
-import org.apache.flink.streaming.api.functions.ProcessFunction;
-import org.apache.flink.util.Collector;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.ENRICHER_CONFIGS_KEY;
+
+import com.typesafe.config.Config;
+import java.util.Map;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
 import org.hypertrace.core.datamodel.StructuredTrace;
 import org.hypertrace.entity.data.service.client.DefaultEdsClientProvider;
 import org.hypertrace.traceenricher.enrichment.EnrichmentProcessor;
 import org.hypertrace.traceenricher.enrichment.EnrichmentRegistry;
 
-/**
- * The Processor function to enrich a StructuredTrace.
- */
-public class StructuredTraceEnrichProcessor extends
-    ProcessFunction<StructuredTrace, StructuredTrace> {
-
-  private static EnrichmentProcessor processor = null;
-
-  public StructuredTraceEnrichProcessor(EnrichmentRegistry registry) {
-    if (processor == null) {
-      synchronized (StructuredTraceEnrichProcessor.class) {
-        if (processor == null) {
-          processor = new EnrichmentProcessor(registry.getOrderedRegisteredEnrichers(),
-              new DefaultEdsClientProvider());
-        }
-      }
-    }
+public class StructuredTraceEnrichProcessor implements
+    Transformer<String, StructuredTrace, KeyValue<String, StructuredTrace>> {
+
+  private EnrichmentProcessor processor;
```

Comment:
The processor was a singleton instance at JVM level earlier, whereas now there will be multiple instances within JVM, one instance per Stream task (i.e. one per partition). Are there any resource/perf considerations due to this change (e.g. multiple EDS cache)?

Reply:
@satish-mittal good point, we may need to take care of this.

Reply:
Done